### PR TITLE
Allow to pass aws.Config

### PR DESCRIPTION
**Pull request by bewiwi**
This PR allow to pass AWS config in the Updater struct to add the possibility to use aws credential not in environment

### CODE REVIEW

File: s3update.go [hunk 1]
```diff
@@ -89,7 +91,11 @@ func runAutoUpdate(u Updater) error {
 		return fmt.Errorf("invalid local version")
 	}
 
-	svc := s3.New(session.New(), &aws.Config{Region: aws.String(u.S3Region)})
+	if u.AWSConfig != nil {
+		u.AWSConfig = &aws.Config{Region: aws.String(u.S3Region)}
+	}
+	svc := s3.New(session.New(), u.AWSConfig)
```
Comment (WinterNis) [inline s3update.go:97]: Is it safe to pass a aws config that is `nil` ? 🤔

Comment (bewiwi) [reply s3update.go:97]: oups wrong condition :smile:

Comment (bewiwi) [reply s3update.go:97]: fixed

**Comment by bewiwi** [comment]
> Sorry to be picky but I think it would be better to still override the region with `u.S3Region` even if `AWSConfig` is set. No ?

don't be sorry, you totally right, I updated it to use only credential.Credentials instead of aws.Config

**Review by WinterNis — APPROVED** [review]

**Pull request merged**